### PR TITLE
ResolveEnv

### PR DESCRIPTION
**Pull request by hculea**

### CODE REVIEW

File: output/secrethub.i [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 extern char *Read(char* path, char** errMessage);
 extern char *Resolve(char* path, char** errMessage);
+extern char *ResolveEnv(char** errMessage)
```
Comment (mackenbach) [inline output/secrethub.i]: You seem to be missing a semi colon `;` at the end 😉

**Comment by florisvdg** [comment]
Could also be done later, but thought it would be good to mention anyway: We want to have this sweet functionality in the Go client too. Would be a shame if every client in every language has it, except for the Go SDK! 😅

**Comment by mackenbach** [comment]
Haha yes please open a PR on the Golang client for `Resolve` and `ResolveEnv(os.Environ())`

File: output/secrethub.i (hunk 1 above)
Comment (edif2008) [reply output/secrethub.i]: Solved

**Comment by edif2008** [comment]
I made a PR which contains the two new functions for the SDK: https://github.com/secrethub/secrethub-go/pull/221

File: secrethub_wrapper.go [hunk 2]
```diff
@@ -116,18 +118,24 @@ func Resolve(ref *C.char, errMessage **C.char) *C.char {
...
+	}
+	encoding, err := json.Marshal(resolvedEnv)
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return nil
 	}
-	return resolvedEnv
-}*/
+	return C.CString(encoding)
+}
```
Comment (Marton6) [inline secrethub_wrapper.go]: This function should call the `ResolveEnv` method implemented on the Go client.

**Review by Marton6 — CHANGES_REQUESTED** [review]

File: secrethub_wrapper.go (hunk 2 above)
Comment (edif2008) [reply secrethub_wrapper.go]: Adjusted. Now the call to `secrethub-go` SDK is made.

Comment (Marton6) [reply secrethub_wrapper.go]: As you know, we decided to postpone adding this functionality to the go client. I'll revert these changes (sorry about that).

File: Makefile [hunk 3]
```diff
@@ -32,12 +32,15 @@ compile-win: $(DEPS)
...
+dotnet-test: TEST=secrethub://secrethub-xgo/dotnet/test-secret
+dotnet-test: OTHER_TEST=secrethub://secrethub-xgo/dotnet/other-test-secret
 .PHONY: dotnet-test
 dotnet-test: lib lib-win
 	cp $(addprefix $(DOTNET_DIR)/, $(TEST_FILES)) $(DOTNET_DIR)/test
 	dotnet publish $(DOTNET_DIR)/test/secrethub.csproj -o $(DOTNET_DIR)/build --nologo
-	mv $(DOTNET_DIR)/libClient.so $(DOTNET_DIR)/build 
-	dotnet test $(DOTNET_DIR)/build/secrethub.dll --nologo
+	mv $(DOTNET_DIR)/libClient.so $(DOTNET_DIR)/build
+	TEST='$(TEST)' OTHER_TEST='$(OTHER_TEST)' dotnet test $(DOTNET_DIR)/build/secrethub.dll --nologo
```
Comment (Marton6) [inline Makefile:43]: I couldn't find a better way to set these variables in the .NET code and make sure that they are also set in the xgo client. This also makes it quite difficult to write a bad weather test case for the ResolveEnv method. I suggest we improve this at a later time.

**Review by Marton6 — APPROVED** [review]

**Pull request merged**